tests/unit/plugins/action/test_deepsec_firewall_rules.py

```diff
@@ -0,0 +1,304 @@
+                    "action": "deny",
+                    "priority": 0,
+                    "source_iptype": "any",
+                    "destination_iptype": "any",
+                    "direction": "outgoing",
+                    "protocol": "tcp",
+                    "log_disabled": True,
+                }
+            ],
+        }
+        result = self._plugin.run(task_vars=self._task_vars)
+        self.assertTrue(result["changed"])
+
+    @patch("ansible.module_utils.connection.Connection.__rpc__")
+    def test_deepsec_firewall_rules_replaced_idempotent(self, connection):
+        self._plugin._connection.socket_path = (
+            tempfile.NamedTemporaryFile().name
+        )
+        self._plugin._connection._shell = MagicMock()
+        self._plugin.search_for_resource_name = MagicMock()
+        self._plugin.search_for_resource_name.return_value = {
+            "firewallRules": [
+                {
+                    "action": "deny",
+                    "priority": "0",
+                    "direction": "outgoing",
+                    "description": "outgoing firewall 1 replaced rule",
+                    "frameType": "ip",
+                    "frameNumber": 2048,
+                    "frameNot": False,
+                    "protocol": "tcp",
+                    "protocolNot": False,
+                    "sourceIPType": "any",
+                    "sourceIPNot": False,
+                    "sourceMACType": "any",
+                    "sourceMACNot": False,
+                    "sourcePortType": "any",
+                    "sourcePortNot": False,
+                    "destinationIPType": "any",
+                    "destinationIPNot": False,
+                    "destinationMACType": "any",
+                    "destinationMACNot": False,
+                    "destinationPortType": "any",
+                    "destinationPortNot": False,
+                    "anyFlags": True,
+                    "logDisabled": True,
+                    "includePacketData": False,
+                    "alertEnabled": False,
+                    "ID": 147,
+                    "name": "test_firewallrule_1",
+                }
+            ]
+        }
+        self._plugin.api_return = "firewallRules"
+        self._plugin._task.args = {
+            "state": "replaced",
+            "config": [
+                {
+                    "name": "test_firewallrule_1",
+                    "description": "outgoing firewall 1 replaced rule",
+                    "action": "deny",
+                    "priority": 0,
+                    "source_iptype": "any",
+                    "destination_iptype": "any",
+                    "direction": "outgoing",
+                    "protocol": "tcp",
+                    "log_disabled": True,
+                }
+            ],
+        }
+        result = self._plugin.run(task_vars=self._task_vars)
+        self.assertFalse(result["changed"])
```

Comment:
All the assertions seem to only check if the task changed or not. Is that expected?